### PR TITLE
fix/AUT-1134/character-restrictions

### PR DESCRIPTION
**Pull request by meefox**
Related to https://oat-sa.atlassian.net/browse/AUT-1134

**Description:**
Add runtime validation for discard invalid symbols

**Changes:**
* Added runtime validation by removing characters from the string

**How to check:**
* Create a new item
* Add inline interaction
* Add inputs with Response base type integer and float
* Click on the preview
* Try to enter characters
* For floats delimiter based by "." due to compatibility

### CODE REVIEW

File: src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js [hunk 1]
```diff
@@ -36,12 +36,51 @@ import tooltip from 'ui/tooltip';
...
-var hideTooltip = function hideTooltip($input) {
+const hideTooltip = function hideTooltip($input) {
     if ($input.data('$tooltip')) {
         $input.data('$tooltip').hide();
     }
 };
 
+const DELIMETER_SYMBOL = '.';
+const MINUS_SYMBOL = '-';
+const cleanupIntegerRegExp = new RegExp('\\D', 'g') // Remove all non-digits except the delimeter
```
Comment (shaveko) [inline src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js:47]: What about e notation? like `5000` === `5e3` both are valid integers. And there is possibility to setup base, that could be 16 so `0x1A` would be also valid input for int

**Review by shaveko — CHANGES_REQUESTED** [review]
I don't think regex is correct approach here.. Can't we rely on parseInt / parseFloat somehow?

**Review by Silvia-loza — APPROVED** [review]
Fix works fine and it looks OK to me, however maybe worth to re-think a bit with @shaveko 's comments in mind...?
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X]  Pull request's target is not` master`
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Comment by meefox** [comment]
Closed no need to merge

**Pull request closed without merging**